### PR TITLE
[build] fix issue when executing `build-commissioner-libs.sh` multiple times

### PR DESCRIPTION
**Pull request by ZhangLe2016**
When developers run the 'build-commissioner-libs.sh' script multiple times, they might see the following error during compiling the libevent:

error: call to undeclared function 'pipe2'; ISO C99 and later do not support implicit function declarations ...

In fact, we did not use pip2 in libevent. We can turn it off by using the macro EVENT_HAVE-PIPE2=0

### CODE REVIEW

**Comment by ZhangLe2016** [comment]
@jwhui seems that I cannot add reviewers into this PR, could you please add me into this project as a contributor? thanks.

**Comment by wgtdkp** [comment]
Better add this to third_party/libevent/CMakeLists.txt:

```
set(EVENT__HAVE_PIPE2 0 CACHE BOOL "disable use of pipe2 in libevent" FORCE)
```

File: android/build-commissioner-libs.sh [hunk 1]
```diff
@@ -55,6 +55,7 @@ cmake -GNinja \
     -DCMAKE_CXX_STANDARD=11 \
     -DCMAKE_CXX_STANDARD_REQUIRED=ON \
     -DCMAKE_BUILD_TYPE=Release \
+    -DEVENT__HAVE_PIPE2=0 \
```
Comment (wgtdkp) [inline android/build-commissioner-libs.sh:58]: This can be removed now

Comment (ZhangLe2016) [reply android/build-commissioner-libs.sh]: Oops! Thanks for the reminder.

**Review by wgtdkp — APPROVED** [review]
LGTM, Thanks! :+1:

**Review by librasungirl — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**